target/ppc/fpu_helper.c

```diff
@@ -2751,6 +2751,35 @@ VSX_CVT_FP_TO_FP_HP(xscvhpdp, 1, float16, float64, VsrH(3), VsrD(0), 1)
 VSX_CVT_FP_TO_FP_HP(xvcvsphp, 4, float32, float16, VsrW(i), VsrH(2 * i  + 1), 0)
 VSX_CVT_FP_TO_FP_HP(xvcvhpsp, 4, float16, float32, VsrH(2 * i + 1), VsrW(i), 0)
 
+void helper_XVCVSPBF16(CPUPPCState *env, ppc_vsr_t *xt, ppc_vsr_t *xb)
+{
+    ppc_vsr_t t = { };
+    int i;
+
+    helper_reset_fpstatus(env);
+    for (i = 0; i < 4; i++) {
+        t.VsrH(2*i + 1) = float32_to_bfloat16(xb->VsrW(i), &env->fp_status);
+        if (unlikely(float32_is_signaling_nan(xb->VsrW(i), &env->fp_status))) {
+            float_invalid_op_vxsnan(env, GETPC());
+            t.VsrH(2*i + 1) = float32_to_bfloat16(
+                float32_snan_to_qnan(xb->VsrW(i)), &env->fp_status);
+        }
+    }
+
+    *xt = t;
+    do_float_check_status(env, GETPC());
+}
+
+void helper_XVCVBF16SPN(CPUPPCState *env, ppc_vsr_t *xt, ppc_vsr_t *xb)
+{
+    int i;
+
+    for (i = 0; i < 4; i++) {
+        xt->VsrH(2*i) = xb->VsrH(2*i + 1);
+        xt->VsrH(2*i + 1) = 0;
+    }
```

Comment:
It seems that we can implement this with GVec... Something like
```c
static void gen_xvcvbf16spn_vec(unsigned vece, TCGv_vec t, TCGv_vec b)
{
    TCGv_vec mask = tcg_temp_new_vec_matching(t);
    tcg_gen_dupi_vec(vece, mask, 0xFFFF0000);
    tcg_gen_shli_vec(vece, t, b, 16);
    tcg_gen_and_vec(vece, t, t, mask);
    tcg_temp_free_vec(mask);
}
```
or something using `tcg_gen_gvec_*` if we can avoid the allocation of `mask`.

Reply:
Just documenting that we went with `tcg_gen_gvec_shli` and the updated version is on e47c7a5